pkg/northbound/gnmi/set.go

```diff
@@ -171,7 +176,62 @@ func (s *Server) doDelete(u *gnmi.Path, targetRemoves mapTargetRemoves) []string
 
 }
 
-func (s *Server) buildUpdateResults(targetUpdates mapTargetUpdates,
+func (s *Server) checkForReadOnly(deviceType string, version string, targetUpdates mapTargetUpdates,
+	targetRemoves mapTargetRemoves) error {
+	configs := manager.GetManager().ConfigStore.Store
+
+	targetModelTypes := make(map[string][]string)
+	for t := range targetUpdates { // map - just need the key
+		if _, ok := targetModelTypes[t]; ok {
```

Comment:
I don't get the meaning of this for loop. if it's there we continue ?

Reply:
It Iterates through all the updates - many may use the same target - here we create a map of the models for all of the targets. Same for the updates. It's just an optimization on the looping

Comment added to code